plugins/output/adapt/outputHelpers.js

```diff
@@ -277,6 +277,43 @@ function getPluginFrameworkVersionCategory(serverFrameworkVersion, pluginMetaDat
   });
 };
 
+function validateCourse(data, cb) {
+  let errors = '';
+  let contentObjects = data.contentobject;
+  let articles = data.article;
+  let blocks = data.block;
+  let components = data.component;
+
+  if (typeof contentObjects === 'undefined') {
+    let courseString = app.polyglot.t('app.course').charAt(0).toUpperCase() + app.polyglot.t('app.course').slice(1);
+    errors += courseString + ' "' + data.course[0].title + '" ' + app.polyglot.t('app.doesnotcontain') + ' ' + app.polyglot.t('app.page') + 's\n';
```

Comment:
Localising pluralised words is unfortunately not as easy as appending an 's'. For example, in French the singular form of courses is _cours_.

The way round it in polyglot is by providing both forms in the language file. For example:
```js
const courseString = app.polyglot.t('app.courses', 1);
errors += app.polyglot.t('app.doesnotcontain', {
  type: courseString[0].toUpperCase() + courseString.slice(1),
  title: data.course[0].title,
  childType: app.polyglot.t('app.pages', 0)
});
```
with language strings:
```json
"app.courses": "course |||| courses",
"app.pages": "page |||| pages",
"app.doesnotcontain": "%{type} '%{title}' does not contain any %{childType}"
```

Reply:
Re testing, I tested all the combos, eg block without component, article without block etc. just to check the error messages. Only slightly weird one was if you have an empty menu the message was 

Menu "Section title" does not contain any articles

But I didn't add a special case to change articles to pages because it seemed self explanatory (and you can't add an article without a page anyway).

Great point about the string localisation, I will add this in!

Thanks for prompt :eyes: btw

Reply:
Updated - I'll leave this conversation for you to resolve if you're happy